File: unreal-common/src/main/java/org/unreal/cloud/common/vo/BaseResponse.java
package org.unreal.cloud.common.vo;

public class BaseResponse<T> {

    private int status = 200;

    private T data;

    public BaseResponse(T data) {
        this.data = data;
    }

    public int getStatus() {
        return status;
    }

    public T getData() {
        return data;
    }
}

File: unreal-common/src/main/java/org/unreal/cloud/common/vo/BaseResponse.kt
package org.unreal.cloud.common.vo

class BaseResponse<T> {
    var status = 200
    var data = "" as T

    constructor()

    private constructor(data: T) {
        this.data = data
    }

    companion object {
        fun <T> ok(data: T): BaseResponse<T> {
            return BaseResponse(data)
        }

        fun ok(): BaseResponse<String> {
            return BaseResponse()
        }
    }
}
